pkg/pillar/zedbox/zedbox.go

```diff
@@ -32,45 +37,155 @@ import (
 	"github.com/lf-edge/eve/pkg/pillar/cmd/zedagent"
 	"github.com/lf-edge/eve/pkg/pillar/cmd/zedmanager"
 	"github.com/lf-edge/eve/pkg/pillar/cmd/zedrouter"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
 	"github.com/lf-edge/eve/pkg/pillar/pubsub"
 	"github.com/lf-edge/eve/pkg/pillar/pubsub/socketdriver"
+	log "github.com/sirupsen/logrus"
 )
 
-var entrypoints = map[string]func(*pubsub.PubSub){
-	"client":           client.Run,
-	"command":          command.Run,
-	"diag":             diag.Run,
-	"domainmgr":        domainmgr.Run,
-	"downloader":       downloader.Run,
-	"executor":         executor.Run,
-	"hardwaremodel":    hardwaremodel.Run,
-	"ledmanager":       ledmanager.Run,
-	"logmanager":       logmanager.Run,
-	"nim":              nim.Run,
-	"nodeagent":        nodeagent.Run,
-	"verifier":         verifier.Run,
-	"volumemgr":        volumemgr.Run,
-	"waitforaddr":      waitforaddr.Run,
-	"zedagent":         zedagent.Run,
-	"zedmanager":       zedmanager.Run,
-	"zedrouter":        zedrouter.Run,
-	"ipcmonitor":       ipcmonitor.Run,
-	"baseosmgr":        baseosmgr.Run,
-	"wstunnelclient":   wstunnelclient.Run,
-	"conntrack":        conntrack.Run,
-	"tpmmgr":           tpmmgr.Run,
-	"vaultmgr":         vaultmgr.Run,
-	"upgradeconverter": upgradeconverter.Run,
-}
+const (
+	agentName   = "zedbox"
+	socketPath  = "/var/run"
+	errorTime   = 3 * time.Minute
+	warningTime = 40 * time.Second
+)
+
+var (
+	debugOverride = false
+	//Version Set from Makefile
+	Version     = "No version specified"
+	entrypoints = map[string]func(*pubsub.PubSub){
+		"client":           client.Run,
+		"command":          command.Run,
+		"diag":             diag.Run,
+		"domainmgr":        domainmgr.Run,
+		"downloader":       downloader.Run,
+		"executor":         executor.Run,
+		"hardwaremodel":    hardwaremodel.Run,
+		"ledmanager":       ledmanager.Run,
+		"logmanager":       logmanager.Run,
+		"nim":              nim.Run,
+		"nodeagent":        nodeagent.Run,
+		"verifier":         verifier.Run,
+		"volumemgr":        volumemgr.Run,
+		"waitforaddr":      waitforaddr.Run,
+		"zedagent":         zedagent.Run,
+		"zedmanager":       zedmanager.Run,
+		"zedrouter":        zedrouter.Run,
+		"ipcmonitor":       ipcmonitor.Run,
+		"baseosmgr":        baseosmgr.Run,
+		"wstunnelclient":   wstunnelclient.Run,
+		"conntrack":        conntrack.Run,
+		"tpmmgr":           tpmmgr.Run,
+		"vaultmgr":         vaultmgr.Run,
+		"upgradeconverter": upgradeconverter.Run,
+	}
+)
 
 func main() {
+	log.Infof("Starting %s", agentName)
+	versionPtr := flag.Bool("v", false, "Version")
+	debugPtr := flag.Bool("d", false, "Debug flag")
+
 	ps := pubsub.New(&socketdriver.SocketDriver{})
+	var sktData string
+	var cmdAndArgs []string
+	sktChan := make(chan string)
+	stillRunning := time.NewTicker(15 * time.Second)
+	dumpHeapMemory := time.NewTicker(5 * time.Second)
```

Comment:
I assume we don't want this periodic dump in the product.
Can we make SIGUSR2 log the profile so we can get it in Kibana? Or do we only want this to be dumped locally on demand in the product?

Reply:
Removed this considering memory usage.